src/openagi/agent.py

```diff
@@ -420,12 +439,78 @@ def single_agent_execution(self, query: str, description: str, task_lists: TaskL
         logging.debug(f"Execution Completed for Session ID - {self.memory.session_id}")
         return output
 
+
     def run(self, query: str, description: str,planned_tasks: Optional[List[Dict]] = None):
         logging.info("Running Admin Agent...")
         logging.info(f"SessionID - {self.memory.session_id}")
 
-        if planned_tasks is None:
-            planned_tasks = self.run_planner(query=query, descripton=description)
+        if self.memory.long_term and planned_tasks:
+            logging.warning("Long Term Memory is not applicable for user given plan.")
+
+        """
+        ltm_context = ""
+        bad_feedback = False
+        bad_session = None
+        if ltm enabled:
+            retrieve ltm
+            for sessions in ltm:
+                if any sim(session, query) > threshold 
+                    if no negative feedback:
+                        ans = session.answer
+                        take feedback 
+                        update session with new feedback  ##UPDATION
+                        return ans
+                    else:
+                        ltm_context += session
+                        bad_feedback = True
+                        bad_session = session
+                        break
+                ltm_context += session
+        
+        planned_tasks = run_planner(query, description, ltm_contText)
+        
+        ans = execute(planned_tasks)
+        
+        if ltm enabled:
+            if bad_feedback:
+                update bad_session with new feedback  ##UPDATION
+            else:
+                save session in ltm                   ##ADD
+        return ans
+        """
+        ltm = ["None"]
+        bad_feedback = False
+        bad_session = None
+        if self.memory.long_term and not planned_tasks:
+            logging.info("Retrieving similar queries from long term memory...")
+            similar_sessions = self.memory.get_ltm(query)
+            ltm = []
+            for memory in similar_sessions:
+                metadata = memory["metadata"]
+                print(memory["similarity_score"])
```

Comment:
use logging instead of print statement. include the similarity score along with the metadata into the logs to verify what was retrieved.

Reply:
added into logging, removed print statement